core/integration/module_test.go

```diff
@@ -1189,6 +1189,113 @@ func (m *Foo) MyFunction() X {
 	require.JSONEq(t, `{"foo":{"myFunction":{"message":"foo"}}}`, out)
 }
 
+func TestModuleGoReturnStruct(t *testing.T) {
+	t.Parallel()
+
+	c, ctx := connect(t)
+
+	modGen := c.Container().From(golangImage).
+		WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+		WithWorkdir("/work").
+		With(daggerExec("mod", "init", "--name=foo", "--sdk=go")).
+		WithNewFile("main.go", dagger.ContainerWithNewFileOpts{
+			Contents: `package main
+
+type Foo struct {}
+
+type X struct {
+	Message string ` + "`json:\"message\"`" + `
+	To string ` + "`json:\"recipient\"`" + `
+	From string
```

Comment:
In addition to these cases, can we also add coverage for a case like
```
Foo string `json:"Bar"`
```

where we expect the original name to be different from the actual graphql field name (`Bar` is original name, `bar` is field name).

I'm pretty sure it works but good to have coverage for.